File: C/HelloWorld.c
#include <stdio.h>
#include <string.h>

struct Dog {
   char name[50];
   int age;
};

void bark(struct Dog dog){
   printf("%s is barking", dog.name);
}

int main() {
   // printf() displays the string inside quotation
   struct Dog sparky;
   strcpy (sparky.name, "Sparky");
   sparky.age = 5;
   printf("%s \n", sparky.name );
   printf("%d \n", sparky.age );
   bark(sparky);
}

// gcc -o test test.c && ./test

File: go/HelloWorld.go
package main

import "fmt"

//Define a struct
type Dog struct {
	name string
	age int
}

// Struct Method
func (dog Dog) bark(){
	fmt.Printf(dog.name + " barks.")
}

func main(){
	sparky := Dog{"Sparky", 5}
	sparky.bark()
}